tools/meteor-services/deploy.js

```diff
@@ -321,6 +334,122 @@ function canonicalizeSite(site) {
+// Default pollInterval is 700 milliseconds – this is the wait interval between polls
+// Default timeout is 15 minutes
+class PollingState {
+  constructor(timeoutMs,
+    initialWaitTimeMs,
+    pollIntervalMs) {
+      const fifteenMinutesMs = 15*60*1000;
+      this.initialWaitTimeMs = initialWaitTimeMs || 10*1000;
+      this.pollIntervalMs = pollIntervalMs || 700;
+      this.timeoutMs = timeoutMs || fifteenMinutesMs
+  }
+
+  // Declare tracking variables for the duration of the poll
+  // `start` tracks the time when we started polling
+  // `currentMessage` tracks what the current status message is for this version
+  initialize() {
+    this.start = new Date();
+    this.currentMessage = '';
+  }
+}
+
+// Poll the "version-status" endpoint for the build and deploy status
+// of a specified version ID with a polling configuration.
+// This will only end successfully when the polling endpoint reports that
+// the version deployment is finished. The version-status endpoints will report
+// messages pertaining to the status of the version, which will then be reported
+// directly to the user. When the poll is complete, it will return an object
+// with information about the final state of the version and the app.
+function pollForDeploy(pollingState, versionId, firstRun = true) {
+  const {
+    timeoutMs,
+    initialWaitTimeMs,
+    pollIntervalMs,
+    start,
+    currentMessage,
+  } = pollingState;
+  // If this is the first poll, we will wait for initialWaitTimeMs milliseconds 
+  // before beginning. We also want to initialize all of our tracking variables
+  if(firstRun) {
+    // Initialize the state of the poll, then begin polling after a pause
+    pollingState.initialize();
+    sleepMs(initialWaitTimeMs);
+    return pollForDeploy(pollingState, versionId, false);
+  }
+
+  // Do a call to the version-status endpoint for the specified versionId
+  const versionStatusResult = deployRpc({
+    method: 'GET',
+    operation: 'version-status',
+    site: versionId,
+    expectPayload: [],
+    printDeployURL: false,
+  });
+  
+  //Check the details of the Version Status response and compare message to last call
+  if(versionStatusResult &&
+    versionStatusResult.payload &&
+    versionStatusResult.payload.message) {
+      const message = versionStatusResult.payload.message;
+      if ( currentMessage !== message ) {
+        Console.info(message);
+        pollingState.currentMessage = message;
+      }
+  } else {
+    // If we did not get a valid Version Status response, just fail silently and
+    // keep polling as per usual – this may have just been a whiff from Galaxy
+    // We do the retry here because we might hit an error if we try to parse the
+    // result of the version-status call below.
+    if(new Date().getTime - start < timeoutMs) {
+      sleepMs(pollIntervalMs);
+      return pollForDeploy(pollingState, versionid, false);
```

Comment:
typo in versionid — needs a capital I, so this would crash. another thing where a linter might help? or just an editor that shows unknown variables in a different color?

Reply:
Yeah VSCode and Intellij don't play well with meteor's code base for me... :/